code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -48,6 +48,7 @@
 /obj/machinery/reagentgrinder/Destroy()
 	if(beaker)
 		beaker.forceMove(drop_location())
+		beaker = null
```

Comment:
Destroy() should not be performing any side effects, this should delete the beaker (QDEL_NULL) and the dropping nonsense should be in deconstruct.